bionic/core/task_execution.py

```diff
@@ -59,6 +59,7 @@ def __init__(self, state):
         self.state = state
         self.future = None
         self.results_by_dnode = None
+        self._stage = None
         self.stage = EntryStage.COMPLETED
```

Comment:
You don't need this anymore.

Reply:
I think we do, because when we set `stage` on the next line, the implementation checks the value of `_stage`.

Reply:
But won't `self.stage` be overridden by the property you defined later? Did you mean to initialize `_stage` with `COMPLETED` or is this some python magic where you can initialize a property this way?

Reply:
By the time we call `__init__`, all the methods defined in the class already exist, so when we do `self.stage = EntryStage.COMPLETED`, that calls the property setter.